Userland/Libraries/LibGfx/Bitmap.cpp

```diff
@@ -476,7 +476,7 @@ RefPtr<Gfx::Bitmap> Bitmap::scaled(float sx, float sy) const
 
 RefPtr<Gfx::Bitmap> Bitmap::cropped(Gfx::IntRect crop) const
 {
-    auto new_bitmap = Gfx::Bitmap::create(format(), { crop.width(), crop.height() }, 1);
+    static auto new_bitmap = Gfx::Bitmap::create(format(), { crop.width(), crop.height() }, 1);
```

Comment:
That doesn't look correct. For one this would return the same bitmap even if - at a later point - a different IntRect was specified.

Also, this seems unnecessary because `new_bitmap` is a heap-allocated object that's wrapped in a smart pointer (RefPtr).

Instead you should change `auto* bitmap` to `auto bitmap` in `shot.cpp` because that would otherwise destroy the RefPtr.

Reply:
Thanks for the correction! 

Can you help me understand why `auto* bitmap` would destroy `RefPtr`?

Reply:
`Gfx::Bitmap::create` returns a `RefPtr<Gfx::Bitmap>`. When you use `auto*` that causes C++ to try to coerce whatever object you specified to the right of `=` into a pointer type.

The `RefPtr<T>` class has this operator which is then implicitly used to coerce the `RefPtr<Gfx::Bitmap>` into a `Gfx::Bitmap*`: `ALWAYS_INLINE operator T*() { return as_ptr(); }`

However, at that point the `RefPtr<Gfx::Bitmap>` that was returned by the `create()` function is no longer necessary and is implicitly destroyed (because you're not holding on to it anywhere) - which in turn also destroys the underlying `Gfx::Bitmap` object - leaving you with a dangling pointer.

Reply:
Thanks for the detailed info! Everything is much clear now.